test/Microsoft.ML.Tests/TrainerEstimators/OneDalEstimators.cs

```diff
@@ -0,0 +1,170 @@
+            Environment.SetEnvironmentVariable("MLNET_BACKEND", "ONEDAL");
+            var trainDataPath = GetDataPath("binary_synth_data_train.csv");
+            var testDataPath = GetDataPath("binary_synth_data_test.csv");
+
+            var loader = ML.Data.CreateTextLoader(columns: new[] {
+              new TextLoader.Column("f0", DataKind.Single, 0),
+              new TextLoader.Column("f1", DataKind.Single, 1),
+              new TextLoader.Column("f2", DataKind.Single, 2),
+              new TextLoader.Column("f3", DataKind.Single, 3),
+              new TextLoader.Column("f4", DataKind.Single, 4),
+              new TextLoader.Column("f5", DataKind.Single, 5),
+              new TextLoader.Column("f6", DataKind.Single, 6),
+              new TextLoader.Column("f7", DataKind.Single, 7),
+              new TextLoader.Column("target", DataKind.Boolean,8)},
+              separatorChar: ',',
+              hasHeader: true
+             );
+
+            var trainingData = loader.Load(trainDataPath);
+            var testingData = loader.Load(testDataPath);
+
+            var preprocessingPipeline = ML.Transforms.Concatenate("Features", new string[] { "f0", "f1", "f2", "f3", "f4", "f5", "f6", "f7" });
+
+            var preprocessedTrainingData = preprocessingPipeline.Fit(trainingData).Transform(trainingData);
+            var preprocessedTestingData = preprocessingPipeline.Fit(trainingData).Transform(testingData);
+
+            // Output.WriteLine($"**** After preprocessing the data got {preprocessedTrainingData.Schema.Count} columns.");
+
+            FastForestBinaryTrainer.Options options = new FastForestBinaryTrainer.Options();
+            options.LabelColumnName = "target";
+            options.FeatureColumnName = "Features";
+            options.NumberOfTrees = 100;
+            options.NumberOfLeaves = 128;
+            options.MinimumExampleCountPerLeaf = 5;
+            options.FeatureFraction = 1.0;
+
+            var trainer = ML.BinaryClassification.Trainers.FastForest(options);
+            var model = trainer.Fit(preprocessedTrainingData);
+            var trainingPredictions = model.Transform(preprocessedTrainingData);
+            var trainingMetrics = ML.BinaryClassification.EvaluateNonCalibrated(trainingPredictions, labelColumnName: "target");
+            var testingPredictions = model.Transform(preprocessedTestingData);
+            var testingMetrics = ML.BinaryClassification.EvaluateNonCalibrated(testingPredictions, labelColumnName: "target");
+
+            Assert.True(testingMetrics.Accuracy > 0.8);
+        }
+
+
+        //[Fact]
+        //[NativeDependencyFact("MatrixFactorizationNative")]
+        [NativeDependencyFact("OneDalNative")]
+        public void TestDependency()
+        {
+            var currentDir = AppContext.BaseDirectory;
+            Output.WriteLine($"**** Running from directory {currentDir}.");
+
+            var dllDir = AppContext.GetData("NATIVE_DLL_SEARCH_DIRECTORIES").ToString();
```

Comment:
This doesn't work on .NETFramework, but I image you've just added this for debugging purposes.  Make sure to remove it.

Reply:
Will do, this whole test is there on a *very* temporary basis.

Reply:
Commit #e2a60945 removes the test.